File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/PreApplyPowerMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import { AbstractPower } from "../Power/AbstractPower";
import AbstractCreature from "../Unit/AbstractCreature";

export class PreApplyPowerMessage implements IMessage{
    public static readonly Id = "PreApplyPowerMessage";
    constructor(public Power:AbstractPower,public Target:AbstractCreature,public Source:AbstractCreature,public StackAmount:number) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/PreApplyPowerMessage.cs
using ModTheSpire.Power;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///应用能力之前
    public class PreApplyPowerMessage : IEventHandle
    {
        public AbstractCreature Source;
        public AbstractCreature Target;      
        public AbstractPower Power;
        public int Amount;

        public PreApplyPowerMessage(AbstractCreature source, AbstractCreature target, AbstractPower power, int amount)
        {
            Source = source;
            Target = target;
            Power = power;
            Amount = amount;
        }
    }
}